samples/core/tutorials/estimators/wide.ipynb

```diff
@@ -0,0 +1,1334 @@
+        "clear_output()"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "IvY3a9pzcYxH",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "After the model is trained, we can evaluate how good our model is at predicting\n",
+        "the labels of the holdout data:"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "L9nVJEO8cYxI",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "results = model.evaluate(test_inpf)\n",
+        "clear_output()\n",
+        "for key,value in sorted(result.items()):\n",
+        "  print('%s: %0.2f' % (key, value))"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "E0fAibNDcYxL",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "The first line of the final output should be something like\n",
+        "`accuracy: 0.83`, which means the accuracy is 83%. Feel free to try more\n",
+        "features and transformations and see if you can do even better!\n",
+        "\n",
+        "After the model is evaluated, we can use the model to predict whether an individual has an annual income of over\n",
+        "50,000 dollars given an individual's information input.\n",
+        "\n",
+        "Let's look in more detail how the model did:"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "8R5bz5CxcYxL",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "import numpy as np\n",
+        "predict_df = test_df[:20].copy()\n",
+        "\n",
+        "pred_iter = model.predict(\n",
+        "    lambda:easy_input_function(predict_df, label_key='income_bracket',\n",
+        "                               num_epochs=1, shuffle=False, batch_size=10))\n",
+        "\n",
+        "classes = np.array(['<=50K', '>50K'])\n",
+        "pred_class_id = []\n",
+        "for pred_dict in pred_iter:\n",
+        "  pred_class_id.append(pred_dict['class_ids'])\n",
+        "\n",
+        "predict_df['predicted_class'] = classes[np.array(pred_class_id)]\n",
+        "predict_df['correct'] = predict_df['predicted_class'] == predict_df['income_bracket']\n",
+        "\n",
+        "clear_output()\n",
+        "predict_df[['income_bracket','predicted_class', 'correct']]"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "N_uCpFTicYxN",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "If you'd like to see a working end-to-end example, you can download our\n",
+        "[example code](https://github.com/tensorflow/models/tree/master/official/wide_deep/census_main.py)\n",
+        "and set the `model_type` flag to `wide`."
+      ]
+    },
+    {
```

Comment:
From @ispirmustafa:

>L1 L2 is an important tool for linear models.
>
>without them it would be really not complete linear example.
>
>have you seen the mlcc content: https://developers.google.com/machine-learning/crash-course/regularization-for-sparsity/video-lecture